kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c

```diff
@@ -0,0 +1,285 @@
+#include <com/amazonaws/kinesis/video/cproducer/Include.h>
+
+#define DEFAULT_RETENTION_PERIOD            2 * HUNDREDS_OF_NANOS_IN_AN_HOUR
+#define DEFAULT_BUFFER_DURATION             120 * HUNDREDS_OF_NANOS_IN_A_SECOND
+#define DEFAULT_CALLBACK_CHAIN_COUNT        5
+#define DEFAULT_KEY_FRAME_INTERVAL          45
+#define DEFAULT_FPS_VALUE                   25
+#define DEFAULT_STREAM_DURATION             20 * HUNDREDS_OF_NANOS_IN_A_SECOND
+#define SAMPLE_AUDIO_FRAME_DURATION         (20 * HUNDREDS_OF_NANOS_IN_A_MILLISECOND)
+#define SAMPLE_VIDEO_FRAME_DURATION         (HUNDREDS_OF_NANOS_IN_A_SECOND / DEFAULT_FPS_VALUE)
+
+#define NUMBER_OF_H264_FRAME_FILES          403
+#define NUMBER_OF_AAC_FRAME_FILES           582
+
+typedef struct {
+    volatile ATOMIC_BOOL firstVideoFramePut;
+    UINT64 streamStopTime;
+    UINT64 streamStartTime;
+    STREAM_HANDLE streamHandle;
+    CHAR sampleDir[MAX_PATH_LEN + 1];
+} SampleCustomData, *PSampleCustomData;
+
+STATUS readFrameData(PFrame pFrame, PCHAR frameFilePath)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT64 size;
+
+    CHK(pFrame != NULL && frameFilePath != NULL, STATUS_NULL_ARG);
+
+    // Get the size and read into frame
+    CHK_STATUS(readFile(frameFilePath, TRUE, NULL, &size));
+    CHK_STATUS(readFile(frameFilePath, TRUE, pFrame->frameData, &size));
+
+    pFrame->size = (UINT32) size;
+
+CleanUp:
+
+    return retStatus;
+}
+
+PVOID putVideoFrameRoutine(PVOID args)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    PSampleCustomData data = (PSampleCustomData) args;
+    Frame frame;
+    UINT32 fileIndex = 0, frameSize;
+    CHAR filePath[MAX_PATH_LEN + 1];
+    STATUS status;
+    BYTE frameBuffer[200000]; // Assuming this is enough
```

Comment:
Bad assumption. Need to have a proper high-water-mark solution. Also, it's a bad idea to have 200K stack allocation

Reply:
changed to loading all frames into memory during setup.